File: src/main/java/com/ultreon/ultranlang/ast/VarDecl.java
package com.ultreon.ultranlang.ast;

public class VarDecl extends AST {
    private Var varNode;
    private Type typeNode;

    public VarDecl(Var varNode, Type typeNode) {
        this.varNode = varNode;
        this.typeNode = typeNode;
    }

    public Var getVarNode() {
        return varNode;
    }

    public void setVarNode(Var varNode) {
        this.varNode = varNode;
    }

    public Type getTypeNode() {
        return typeNode;
    }

    public void setTypeNode(Type typeNode) {
        this.typeNode = typeNode;
    }
}

File: src/commonMain/kotlin/com/ultreon/ultranlang/ast/VarDecl.kt
package com.ultreon.ultranlang.ast

import com.ultreon.ultranlang.symbol.ClassSymbol
import kotlin.String

class VarDecl(var varRefNode: VarRef, var typeNode: Type, override var classDeclaration: ClassDeclaration? = null, override val isStatic: Boolean = true) : LangObj(), ClassMemberDecl, FieldDecl {
    override var `this`: ClassSymbol? = null
    override val isConstant: Boolean = false
    override val name: String
        get() = varRefNode.value
}